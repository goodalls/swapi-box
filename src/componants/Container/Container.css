.container {
  grid-column: 1/13;
  grid-row: 4;
  height: 50px;
}