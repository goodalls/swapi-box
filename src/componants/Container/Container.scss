.container {
  grid-column: 1/13;
  grid-row: 4;
  height: 50px;
  display: grid;
  grid-template-columns: repeat(5, 1fr);

}