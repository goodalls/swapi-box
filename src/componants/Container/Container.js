import React from 'react';
import PropTypes from 'prop-types';
import './Container.css';
import Card from '../Card/Card';

const Container = props => {
  const favoriteCheck = props.dataArray.length ? 'dataArray' : 'favorites';

  const peopleCards = props[favoriteCheck].map((eachCard, index) => {
    return (
      <Card
        {...eachCard}
        favorite={props.favorite}
        key={index}
        favorites={props.favorites}
      />
    );
  });

  return <div className="container">{peopleCards}</div>;
};

Container.propTypes = {
  favorite: PropTypes.func,
  dataArray: PropTypes.array,
  favorites: PropTypes.array
};

export default Container;
