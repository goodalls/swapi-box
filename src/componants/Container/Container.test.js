import React from 'react';
import Container from './Container';
import { shallow } from 'enzyme';

describe('Container', () => {
  let mockArray;
  let wrapper;

  beforeEach('', () => {
    mockArray = [{}];
    wrapper = shallow(<Container dataArray={mockArray} favorites={mockArray} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should favoriteCheck', () => {
    wrapper = shallow(<Container dataArray={[]} favorites={[{}, {}]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
