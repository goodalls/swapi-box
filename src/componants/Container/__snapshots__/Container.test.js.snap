// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should favoriteCheck 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Container
    dataArray={Array []}
    favorites={
        Array [
            Object {},
            Object {},
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          favorite={undefined}
          favorites={
                    Array [
                              Object {},
                              Object {},
                            ]
          }
/>,
        <Card
          favorite={undefined}
          favorites={
                    Array [
                              Object {},
                              Object {},
                            ]
          }
/>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "favorite": undefined,
          "favorites": Array [
            Object {},
            Object {},
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "favorite": undefined,
          "favorites": Array [
            Object {},
            Object {},
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            favorite={undefined}
            favorites={
                        Array [
                                    Object {},
                                    Object {},
                                  ]
            }
/>,
          <Card
            favorite={undefined}
            favorites={
                        Array [
                                    Object {},
                                    Object {},
                                  ]
            }
/>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "favorite": undefined,
            "favorites": Array [
              Object {},
              Object {},
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "favorite": undefined,
            "favorites": Array [
              Object {},
              Object {},
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Container should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Container
    dataArray={
        Array [
            Object {},
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          favorite={undefined}
          favorites={undefined}
/>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "favorite": undefined,
          "favorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            favorite={undefined}
            favorites={undefined}
/>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "favorite": undefined,
            "favorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
